File: C++/factorial.cpp
#include<iostream>
using namespace std;

 int factorial(int n){
   if(n<=1){
     return 1;
   }
   return n * factorial(n-1);
 }

int main(){
  int n;
  cout<< "enter the number : "<<endl;
  cin>>n;
  cout<<" the fecroriyal is"<<factorial(n);
  return 0;
}

File: java/Factorial.java
public class Factorial {
    public static void main(String[] args) {
        int n=5;
        int i=1;
        int fac = 1;
        while(i<=n){
            fac = fac * i;
            i++;
        }
        System.out.printf("factorial of %d is %d ",n,fac);
    }
    
}
